### ISSUE TITLE
🐞 Missing Function for Replacing Characters with Multiple Ranges and Strings

### ISSUE THREAD

**Issue by activcoding** [CodeEditApp/CodeEditTextView#44]
### Description

The current implementation lacks a function that accepts multiple ranges and corresponding strings for replacement. We need a method similar to the following:
```swift
public func replaceCharacters(in ranges: [[NSRange]], with strings: [String])
```
This function should allow replacing characters in multiple ranges with their respective strings.
While you can call `replaceCharacters(_:with:)` multiple times, each call is treated as a separate insertion, which creates a misleading undo history.
### Example:
Comments:
https://github.com/user-attachments/assets/c7451af9-6d63-49b1-b52d-6e9b020461f2

### To Reproduce

x

### Expected Behavior

x

**Comment by thecoolwinter** [comment]
After starting working on this, I don't think it's a necessary API. If the issue it's solving is undo grouping, you should be able to call [`textView.undoManager?.beginGrouping()`](https://codeeditapp.github.io/CodeEditTextView/documentation/codeedittextview/ceundomanager/begingrouping()) before and [`textView.undoManager?.endGrouping()`](https://codeeditapp.github.io/CodeEditTextView/documentation/codeedittextview/ceundomanager/endgrouping()) after applying all your edits to force the undo manager to group them all into one undo group.

**Comment by activcoding** [comment]
You’re absolutely right; it makes more sense to implement that situation specifically
